### ISSUE TITLE
`type` in json dump should be `@type`

### ISSUE THREAD

**Issue by hsolbrig** [biolink/biolinkml#364]
The JSON serializer emits a `type` variable in the outermost element.  We need to consider:

a) Changing this to `@type` so we don't have any potential collisions w/ LinkML variables
b) Emit it in ALL classes, not just the outermost

Note that one drawback of taking this approach is that it will _force_ `rdf:type` arcs to be emitted on all subjects.  I would propose that we add the ability to produce arcs as an option.  Possibilities could include "none", "top" or "all" (default: "all").

**Comment by TomConlin** [comment]
I do not think all RDF-subjects can be expected to be RDF-Entities which being given a `rdf:type` predicate would imply.  
That is without being able to guarantee every RDF-subject is a member of some Class I think it wise to stay away from 'all'  unless the universe of RDF-subjects we agree to deal with are so constrained.

**Comment by hsolbrig** [comment]
Makes sense.  We find ourselves in the situation where we have two senses of `rdf:type` -- one that indicates what the instance is intended to represent (e.g. `<JohnSmith> a foaf:Person`) and a second about how it is recorded `<JohnSmith> a fhir:Patient`) (replace `fhir` with the model of your chioce).  Q: is there a predicate in IAO or elsewhere that we can use to represent this second sense?

**Comment by TomConlin** [comment]
The concept of "role" comes to mind.
Surely that must be ontologized somewhere
.... yep:

https://www.ebi.ac.uk/ols/search?q=role&groupField=iri&start=0&type=property

Choosing which is likely politics.
